neqo-transport/src/connection.rs

```diff
@@ -611,9 +611,12 @@ impl Connection {
     }
 
     pub fn process_timer(&mut self, now: Instant) {
-        if let State::Closing { error, .. } = self.state().clone() {
-            self.set_state(State::Closed(error));
-            return;
+        match self.state.clone() {
+            State::Closing { error, timeout, .. } if timeout <= now => {
```

Comment:
Can we avoid the clone on the way in and only clone error when we need it?

Reply:
yep